gistUpload.test.js

```diff
@@ -18,27 +18,28 @@ describe('When given no arguments or incorrect number of arguments', () => {
         expect(error.message).toEqual('You must provide a file and a name.');
       });
   });
+});
 
-  it('Returns a new error message: you must provide a file and name.', () => {
-    uploadFile('hello')
-      .catch((error) => {
-        expect(error.message).toEqual('You must provide a file and a name.');
-      });
+describe('When invalid fields are sent', () => {
+  it('Returns error status code 422', () => {
+    axios.post = jest.fn(() => Promise.reject({ response: { status: '422' } }));
+    uploadFile('hello', 'helloName')
+      .then((result) => { expect(result).toBe('422'); });
   });
 });
 
-describe('When given non-string arguments', () => {
-  it('Returns the error: Your file must contain a string.', () => {
-    uploadFile(1)
-      .catch((error) => {
-        expect(error.message).toEqual('Your file must contain a string.');
-      });
+describe('When API limits are reached', () => {
+  it('Returns error status code 403', () => {
+    axios.post = jest.fn(() => Promise.reject({ response: { status: '403' } }));
+    uploadFile('hello', 'helloName')
+      .then((result) => { expect(result).toBe('403'); });
   });
+});
 
-  it('Returns the error: Your gist name must be a string.', () => {
-    uploadFile('hello', 2)
-      .catch((error) => {
-        expect(error.message).toEqual('Your gist name must be a string.');
-      });
+describe('When invalid JSON is sent', () => {
+  it('Returns error status code 400', () => {
+    axios.post = jest.fn(() => Promise.reject({ response: { status: '400' } }));
+    uploadFile('hello', 'helloName')
+      .then((result) => { expect(result).toBe('400'); });
```

Comment:
my suggestion would be to return an object of consistent shape and handle the different cases outside of this function, i.e. on a success you return `{ success: true, message: 'http://foo.com' }` and on an error case you return `{ success: false, message: 'API limit reached' }` or whatever the status message is. this will allow you to switch on the success key to tailor your message to something helpful to the user, like `Gist successfully uploaded!\nYou can see your gist at the following url:\nhttp://foo.com/hash` or `There was an error uploading your gist:\nAPI limit reached` etc.